src/plugins/data/common/data_views/data_views/data_view.ts

```diff
@@ -40,21 +40,26 @@ export class DataView extends IndexPattern implements IDataView {
     });
 
     this.savedObjectsClient = savedObjectsClient;
-    this.initializeDataSourceRef();
   }
 
   public async initializeDataSourceRef(): Promise<void> {
     if (!this.dataSourceRef?.id) return;
-    const dataSourceSavedObject = await this.savedObjectsClient.get(
-      this.dataSourceRef.type,
-      this.dataSourceRef.id
-    );
-    const attributes = dataSourceSavedObject.attributes as any;
-    this.dataSourceRef = {
-      id: this.dataSourceRef.id,
-      type: attributes.dataSourceEngineType || this.dataSourceRef.type,
-      name: attributes.title || this.dataSourceRef.name || this.dataSourceRef.id,
-    };
+
+    try {
+      const dataSourceSavedObject = await this.savedObjectsClient.get(
+        'data-source',
+        this.dataSourceRef.id
+      );
+      const attributes = dataSourceSavedObject.attributes as any;
+      this.dataSourceRef = {
+        id: this.dataSourceRef.id,
+        type: attributes.dataSourceEngineType || this.dataSourceRef.type,
+        name: attributes.title || this.dataSourceRef.name || this.dataSourceRef.id,
+      };
+    } catch (error) {
+      // If data source fetch fails, keep the existing dataSourceRef as-is
```

Comment:
Nit: Seems a little weird to have an error and not do anything in this catch block, is there any logging we could do?